deps/spidershim/src/v8arraybuffer.cc

```diff
@@ -122,6 +122,14 @@ ArrayBufferView::ByteLength()
   return JS_GetDataViewByteLength(view);
 }
 
+void
+ArrayBuffer::Neuter()
+{
+  JSContext* cx = JSContextFromIsolate(Isolate::GetCurrent());
+  JS::RootedObject obj(cx, &reinterpret_cast<JS::Value*>(this)->toObject());
+  JS_DetachArrayBuffer(cx, obj, KeepData);
```

Comment:
On Tue, May 10, 2016 at 10:08:32AM -0700, Ehsan Akhgari wrote:
> In deps/spidershim/src/v8arraybuffer.cc:
> 
> > @@ -122,6 +122,14 @@ ArrayBufferView::ByteLength()
> >    return JS_GetDataViewByteLength(view);
> >  }
> >
> > +void
> > +ArrayBuffer::Neuter()
> > +{
> > +  JSContext* cx = JSContextFromIsolate(Isolate::GetCurrent());
> > +  JS::RootedObject obj(cx, &reinterpret_cast<JS::Value*>(this)->toObject());
> > +  JS_DetachArrayBuffer(cx, obj, KeepData);
> 
> Hmm, what happens to the underlying buffer at this stage? As far as I can tell,
> we'd be leaking it here...

If js owns the data then JS_DetachArrayBuffer() will free it.  However
in v8 it seems to be valid  only to call this on ArrayBuffer's where the
embedder owns the data.  So in the places this should be called it is
intended that the buffer doesn't own the data and so not freeing it is
correct.